run.cc

```diff
@@ -3,80 +3,180 @@
 #include <fstream>
 #include <iostream>
 
+#include "lib/argparse/argparse.h"
 #include "src/includes.h"
 
-int main(int argc, char *argv[]) {
-  std::cout << "Running run.cc" << std::endl;
+bool check_file_is_writeable(std::string filename) {
+  std::ofstream ofstr;
+  ofstr.open(filename);
+  if (ofstr.fail()) {
+    std::cerr << "Error: cannot write to " << filename << std::endl;
+    ofstr.close();
+    return false;
+  }
+  ofstr.close();
+  return true;
+}
 
-  // Get console parameters
-  std::string algo_params_file = argv[1];
-  std::string hier_type = argv[2];
-  std::string hier_args = argv[3];
-  std::string mix_type = argv[4];
-  std::string mix_args = argv[5];
-  std::string collname = argv[6];
-  std::string datafile = argv[7];
-  std::string gridfile = argv[8];
-  std::string densfile = argv[9];
-  std::string nclufile = argv[10];
-  std::string clusfile = argv[11];
-  std::string hier_cov_file;
-  std::string hier_grid_cov_file;
-  std::string mix_cov_file;
-  std::string mix_grid_cov_file;
-  if (argc >= 14) {
-    hier_cov_file = argv[12];
-    hier_grid_cov_file = argv[13];
+bool check_args(argparse::ArgumentParser args) {
+  if (args["--collname"] != std::string("memory")) {
+    check_file_is_writeable(args.get<std::string>("--collname"));
+  }
+  if (args["--gridfile"] != std::string("\"\"")) {
+    check_file_is_writeable(args.get<std::string>("--densfile"));
```

Comment:
That's weird, because setting a filename equal to "" from the Python interface produces the correct result.. Go figure

Reply:
Oooohhh I didn't get it, the file name is `""`, not the empty string! It works if you write `\"\"`, which lets the terminal interpret it correctly as a two-character string rather than an empty one. A bit confusing, but I can get behind it...